pkg/common/config/types.go

```diff
@@ -25,6 +25,9 @@ type Config struct {
 		VCenterIP string
 		// Kubernetes Cluster ID
 		ClusterID string `gcfg:"cluster-id"`
+		// SupervisorID is the UUID representing Supervisor Cluster. ClusterID is being deprecated
+		// and SupervisorID is the replacement ID we need to use for VolumeMetadata and datastore lookup.
```

Comment:
Do you mean "ClusterID" is no longer used for Supervisor Cluster, but it is still used for Vanilla/Guest clusters?

Reply:
from vSphere 8.0, we will have SupervisorID populated and ClusterID is deprecated.
Here in this PR, we are using the new SupervisorID when we find SupervisorID is populated and Cluster has at least one zone CR. For older supervisors created before vSphere 8.0, even if SupervisorID is present in the config secret, we will continue to use cluster-id for volume metadata. This ensures that only new k8s clusters will use the new `supervisor-id` and older k8s will continue to use the `cluster-id`.

For Vanilla and Guest, there is no change. we will use cluster-id.

The change is required in the Supervisor as it is using ClusterMoref as the cluster-id, and now the supervisor is stretched across multiple vSphere Cluster, so this change is required.